src/data_source/extension.rs

```diff
@@ -276,6 +279,62 @@ where
     }
 }
 
+#[async_trait]
+impl<D, U, Types, State> MerklizedStateDataSource<Types, State> for ExtensibleDataSource<D, U>
+where
+    D: MerklizedStateDataSource<Types, State> + Send + Sync,
+    U: Send + Sync,
+    Types: NodeType,
+    State: MerklizedState<Types> + 'static,
+    State::Index: Send + Sync,
+    State::Element: Send + Sync,
+    State::NodeValue: Send,
```

Comment:
Might be cleaner to put these trait bounds on the definitions of the associated types so we don't have to repeat them at every usage